paymentsheet/src/main/java/com/stripe/android/customersheet/data/CustomerSessionElementsSessionManager.kt

```diff
@@ -0,0 +1,98 @@
+
+    private var intentConfiguration: CustomerSheet.IntentConfiguration? = null
+
+    override suspend fun fetchCustomerSessionEphemeralKey(): Result<CachedCustomerEphemeralKey> {
+        return withContext(workContext) {
+            cachedCustomerEphemeralKey.takeUnless { cachedCustomerEphemeralKey ->
+                cachedCustomerEphemeralKey == null || cachedCustomerEphemeralKey.shouldRefresh(
+                    timeProvider()
+                )
+            }?.let {
+                Result.success(it)
+            } ?: run {
+                fetchElementsSession().mapCatching {
+                    cachedCustomerEphemeralKey
+                        ?: throw IllegalStateException("Should have been initialized from `elements/session`!")
+                }
+            }
+        }
+    }
+
+    override suspend fun fetchElementsSession(): Result<ElementsSession> {
+        return withContext(workContext) {
+            runCatching {
+                val intentConfiguration = intentConfiguration
+                    ?: customerSessionProvider.intentConfiguration()
+                        .onSuccess { intentConfiguration = it }
+                        .getOrThrow()
+
+                val customerSessionClientSecret = customerSessionProvider
+                    .providesCustomerSessionClientSecret()
+                    .getOrThrow()
+
+                val prefsRepository = prefsRepositoryFactory(customerSessionClientSecret.customerId)
+
+                val savedSelection = prefsRepository.getSavedSelection(
+                    isGooglePayAvailable = false,
+                    isLinkAvailable = false,
+                ) as? SavedSelection.PaymentMethod
+
+                elementsSessionRepository.get(
+                    initializationMode = PaymentSheet.InitializationMode.DeferredIntent(
+                        intentConfiguration = PaymentSheet.IntentConfiguration(
+                            mode = PaymentSheet.IntentConfiguration.Mode.Setup(),
+                            paymentMethodTypes = intentConfiguration.paymentMethodTypes,
+                        )
+                    ),
+                    defaultPaymentMethodId = savedSelection?.id,
+                    customer = PaymentSheet.CustomerConfiguration.createWithCustomerSession(
+                        id = customerSessionClientSecret.customerId,
+                        clientSecret = customerSessionClientSecret.clientSecret,
+                    ),
+                    externalPaymentMethods = listOf(),
+                ).onSuccess { elementsSession ->
+                    elementsSession.customer?.session?.run {
+                        cachedCustomerEphemeralKey = CachedCustomerEphemeralKey(
+                            customerId = customerId,
+                            ephemeralKey = apiKey,
+                            expiresAt = apiKeyExpiry,
+                        )
+                    }
+                }.getOrThrow()
+            }
+        }
```

Comment:
This function will be used by the `InitializationDataSource` and `PaymentMethodDataSource` which is why it is exposed